android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BroadcastActivity.java

```diff
@@ -0,0 +1,242 @@
+/*
+MainActivity.java
+
+This is the main entry point for the application
+It is the primary ui interface
+It also creates the singletons used for the main app logic
+like sharing a URL via bluetooth, and interfacing with Drive
+It also contains the logic for ensuring permissions
+and other account requirements are satisfied
+*/
+
+package org.physical_web.physicalweb;
+
+import android.Manifest;
+import android.app.Activity;
+import android.accounts.AccountManager;
+import android.app.Dialog;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothManager;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.IBinder;
+import android.preference.PreferenceManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.CompoundButton;
+import android.widget.Switch;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.content.BroadcastReceiver;
+import android.content.IntentFilter;
+    
+// import com.google.android.gms.common.ConnectionResult;
+// import com.google.android.gms.common.GooglePlayServicesUtil;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+
+public class BroadcastActivity extends Activity {
+
+    private static final String TAG  ="BroadcastActivity";
+    private static final int ENABLE_BLUETOOTH_REQUEST_ID = 1;
+    public static final int ACCOUNT_PICKER_REQUEST_ID = 2;
+    public static final int PERMISSIONS_REQUEST_ID = 3;
+    public static final int GOOGLE_PLAY_SERVICES_REQUEST_ID = 5;
+    private HereYouGoService mHereYouGoService;
+    private boolean mAmBoundToHereYouGoService = false;
+    private String ourl;
+
+
+
+    // callbacks
+    /////////////////////////////////
+    @Override
+    protected void onActivityResult (int requestCode, int resultCode, Intent data) {
+      Log.d(TAG,"onActivityResult");
+      if (requestCode == ENABLE_BLUETOOTH_REQUEST_ID) {
+        // Make sure the request was successful
+        Log.d(TAG,resultCode+"");
+        if(resultCode==-1){
+            if (!checkIfRequirement2Fulfilled()) {
+                return;
+            }
+            preBroadcast();
+            finish();
+        }else{
+            Toast.makeText(this, "Turn Bluetooth on to broadcast", Toast.LENGTH_LONG).show();
```

Comment:
Don't we have a better way of doing this right now? (that actually prompts the user and then starts broadcasting when they have accepted?)